src/js/lang.js

```diff
@@ -249,8 +249,91 @@
          * The right delimiter can be [ (exclusive) or ] (inclusive).
          * Beside numbers, you can use -Inf and +Inf for the infinite.
          */
+        
+        var numbers = this._parseNumbersFromInterval(interval);
+        
+        var types = {
+            "setOfNumbers" : /^\{.*\}$/,
+            "bothExclusive": /^(\(|\]|\)).*(\)|\[|\()$/,
+            "bothInclusive": /^\[.*\]$/,
+            "leftInclusive": /^\[.*(\)|\[|\()$/,
+            "rightInclusive": /^(\(|\]|\)).*\]$/
+        };
+
+        if (interval.match(types.setOfNumbers)) {
+            return numbers.indexOf(count) != -1;
+        }
+
+        if (interval.match(types.bothInclusive)) {
+            return count >= numbers[0] && count <= numbers[1];
+        }
+
+        if (interval.match(types.bothExclusive)) {
+            return count > numbers[0] && count < numbers[1];
+        }
+
+        if (interval.match(types.rightInclusive)) {
+            return count > numbers[0] && count <= numbers[1];
+        }
+
+        if (interval.match(types.leftInclusive)) {
+            return count >= numbers[0] && count < numbers[1];
+        }
+    };
+
+    /**
+     * Parse a given string (number, +Inf, -Inf, Inf) to Number.
+     *
+     * @param  {String} str 
+     * @return {Number}     
+     */
+    Lang.prototype._parseNumber = function (str){
+        if (str=='-Inf') {
+            return Math.log(0);
+        }
+
+        if(str=='Inf' || str=='+Inf') {
+            return -Math.log(0);
+        }
+
+        return (new Number(str)).valueOf();
+    };
+
+    /**
+     * Parse an interval to array.
+     * 
+     * @param  {String} interval
+     * @return {Array} 
+     */
+    Lang.prototype._parseNumbersFromInterval = function (interval) {
+        var numbers = interval.split(/,\s?/);
+
+        
+
+        var total = numbers.length;
+
+        var braces = /\[|\]|\{|\}|\(|\)/g;
+
+        /* first number */
+        var first  = numbers[0].replace(braces, '');
+        
+        numbers[0] = first;
+        
+        var last = '';
```

Comment:
@vluzrmos can we put all variable declaration together without new lines separating them? Something like:

```js
var numbers = interval.split(/,\s?/);
var total = numbers.length;
var braces = /\[|\]|\{|\}|\(|\)/g;
var first  = numbers[0].replace(braces, '');
```

Also, I think `var last = '';` can be put together.